utils/s3/credentials_providers/config_file_aws_credentials_provider.hh

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+#include "aws_credentials_provider.hh"
+
+namespace aws {
+
+/**
+ * Reads credentials form yaml file as described in https://github.com/scylladb/scylladb/blob/master/docs/dev/object_storage.md
+ */
+
+class config_file_aws_credentials_provider final : public aws_credentials_provider {
+public:
+    explicit config_file_aws_credentials_provider(const std::string& _creds_file);
+    [[nodiscard]] seastar::future<s3::aws_credentials> get_aws_credentials() override;
+    [[nodiscard]] const char* get_name() const override { return "config_file_aws_credentials_provider"; }
+
+protected:
+    [[nodiscard]] static bool is_time_to_refresh() { return false; }
+    seastar::future<> reload();
+
+private:
+    std::string creds_file;
+    s3::aws_credentials creds;
```

Comment:
This field is never reset, this means, that get_aws_credentials() always returns whatever value is put in it at the first call to this method, so I don't get how changing config and resetting _credentials on s3::client object even helps re-reading the config file